.github/workflows/add-remove-new-fulcio.yaml

```diff
@@ -0,0 +1,221 @@
+
+        FULCIO_URL=$(kubectl -n fulcio-system get ksvc fulcio -ojsonpath='{.status.url}')
+        echo "FULCIO_URL=$FULCIO_URL" >> $GITHUB_ENV
+
+        CTLOG_URL=$(kubectl -n ctlog-system get ksvc ctlog -ojsonpath='{.status.url}')
+        echo "CTLOG_URL=$CTLOG_URL" >> $GITHUB_ENV
+
+        ISSUER_URL=$(kubectl get ksvc gettoken -ojsonpath='{.status.url}')
+        echo "ISSUER_URL=$ISSUER_URL" >> $GITHUB_ENV
+        OIDC_TOKEN=`curl -s $ISSUER_URL`
+        echo "OIDC_TOKEN=$OIDC_TOKEN" >> $GITHUB_ENV
+
+    - name: Sign with cosign from the action using k8s token
+      run: |
+        cosign sign --rekor-url ${{ env.REKOR_URL }} --fulcio-url ${{ env.FULCIO_URL }} --force --allow-insecure-registry ${{ env.demoimage }} --identity-token ${{ env.OIDC_TOKEN }}
+
+    - name: Verify with cosign from the action using k8s token
+      run: |
+        cosign verify --rekor-url ${{ env.REKOR_URL }} --allow-insecure-registry ${{ env.demoimage }}
+
+    - name: Spin up a new Fulcio with new keys
+      run: |
+        ko apply -BRf ./testdata/config/new-fulcio
+        kubectl wait --timeout 5m -n fulcio-system --for=condition=Ready ksvc fulcio-new
+        NEW_FULCIO_URL=$(kubectl -n fulcio-system get ksvc fulcio-new -ojsonpath='{.status.url}')
+        echo "NEW_FULCIO_URL=$NEW_FULCIO_URL" >> $GITHUB_ENV
+
+    - name: Add new Fulcio to ctlog as trusted Fulcio
+      run: |
+        ko apply -BRf ./testdata/config/add-new-fulcio
+        sleep 2
+        kubectl -n ctlog-system wait --for=condition=Complete --timeout=180s job/add-fulcio
+
+    - name: Restart ctlog pod
+      run: |
+        kubectl -n ctlog-system get pods -oname | xargs kubectl -n ctlog-system delete
+        sleep 10
+
+    - name: Dump the trusted certs
+      run: |
+        curl ${{ env.CTLOG_URL }}/sigstorescaffolding/ct/v1/get-roots | jq .certificates
+
+    - name: Verify both Fulcio certs are there
+      run: |
+        go run ./cmd/ctlog/verifyfulcio/main.go \
+        --ctlog-url ${{ env.CTLOG_URL }} \
+        --log-prefix sigstorescaffolding \
+        --fulcio ${{ env.FULCIO_URL }} \
+        --fulcio ${{ env.NEW_FULCIO_URL }}
+
+    - name: Remove old Fulcio from ctlog as trusted Fulcio
+      run: |
+        ko apply -BRf ./testdata/config/remove-old-fulcio
+        sleep 2
+        kubectl -n ctlog-system wait --for=condition=Complete --timeout=180s job/remove-fulcio
+
+    - name: Restart ctlog pod again to pick up the changes
+      run: |
+        kubectl -n ctlog-system get pods -oname | xargs kubectl -n ctlog-system delete
+        sleep 10
+
+    - name: Dump the trusted certs
+      run: |
+        curl ${{ env.CTLOG_URL }}/sigstorescaffolding/ct/v1/get-roots | jq .certificates
+
+    - name: Verify that only new Fulcio cert is there
+      run: |
+        go run ./cmd/ctlog/verifyfulcio/main.go \
+        --ctlog-url ${{ env.CTLOG_URL }} \
+        --log-prefix sigstorescaffolding \
+        --fulcio ${{ env.NEW_FULCIO_URL }}
```

Comment:
Very nice job, love being able to see all of this work composed together

Reply:
Aw, thanks!!!